### ISSUE TITLE
Plugin "azure.azcollection.azure_rm" parameter "hostnames " that allows assigning hostname does not work.

### ISSUE THREAD

**Issue by verodhi** [ansible/ansible_collections_azure#350]
##### SUMMARY

I am using azure.azcollection.azure_rm which is a plugin for dynamic inventory.
It has a filed called "hostnames" that allows us to define the list of hostnames. [Documentation here](https://docs.ansible.com/ansible/latest/collections/azure/azcollection/azure_rm_inventory.html#parameter-hostnames)

However the provided names is not populated when the host is imported in the inventory.
I am running this in AWX 9.2.0 with Ansible 2.9.3 running.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

plugin
##### ANSIBLE VERSION

```paste below
2.9.3
```

##### CONFIGURATION

```paste below
    2.319 INFO     Updating inventory 9: aes_coe_azurevms
    2.777 DEBUG    Using base command: python /usr/bin/ansible-inventory -i /tmp/awx_1087_gan_kil6/project/inventories/azure --playbook-dir /tmp/awx_1087_gan_kil6/project/inventories/azure -vvvvv
    2.777 INFO     Reading Ansible inventory source: /tmp/awx_1087_gan_kil6/project/inventories/azure
    2.779 INFO     Using VIRTUAL_ENV: /var/lib/awx/venv/ansible
    2.779 INFO     Using PATH: /var/lib/awx/venv/ansible/bin:/var/lib/awx/venv/awx/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    2.779 INFO     Using PYTHONPATH: /var/lib/awx/venv/ansible/lib/python3.6/site-packages:
   11.136 ERROR    ansible-inventory 2.9.3
   11.136 ERROR      config file = /tmp/awx_1087_gan_kil6/project/ansible.cfg
   11.136 ERROR      configured module search path = ['/var/lib/awx/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
   11.136 ERROR      ansible python module location = /usr/lib/python3.6/site-packages/ansible
   11.136 ERROR      executable location = /usr/bin/ansible-inventory
   11.136 ERROR      python version = 3.6.8 (default, Nov 21 2019, 19:31:34) [GCC 8.3.1 20190507 (Red Hat 8.3.1-4)]
   11.136 ERROR    Using /tmp/awx_1087_gan_kil6/project/ansible.cfg as config file
   11.136 ERROR    setting up inventory plugins
   11.136 ERROR    host_list declined parsing /tmp/awx_1087_gan_kil6/project/inventories/azure/rightscale.azure_rm.yaml as it did not pass its verify_file() method
   11.136 ERROR    script declined parsing /tmp/awx_1087_gan_kil6/project/inventories/azure/rightscale.azure_rm.yaml as it did not pass its verify_file() method
   11.136 ERROR    /var/lib/awx/venv/ansible/lib/python3.6/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
   11.136 ERROR      InsecureRequestWarning,
   11.136 ERROR    Parsed /tmp/awx_1087_gan_kil6/project/inventories/azure/rightscale.azure_rm.yaml inventory source with auto plugin
   11.136 DEBUG    Finished loading from source: /tmp/awx_1087_gan_kil6/project/inventories/azure
   11.137 INFO     Processing JSON output...
   11.137 DEBUG    Loaded group: all
   11.137 DEBUG    Loaded group: skEnvironmentType_EnvNotDefined
   11.137 DEBUG    Adding child group skEnvironmentType_EnvNotDefined to parent all
   11.137 DEBUG    Loaded host: USERTSTUBUNTU1_2bd7
   11.137 DEBUG    Adding host USERTSTUBUNTU1_2bd7 to group skEnvironmentType_EnvNotDefined
   11.137 DEBUG    Loaded host: USERTSTWINDOWS1_4df5
   11.137 DEBUG    Adding host USERTSTWINDOWS1_4df5 to group skEnvironmentType_EnvNotDefined
   11.137 DEBUG    Adding child group skEnvironmentType_EnvNotDefined to parent all
   11.137 DEBUG    Adding child group skEnvironmentType_EnvNotDefined to parent all
   11.137 DEBUG    Loaded group: os_linux
   11.137 DEBUG    Adding child group os_linux to parent all
   11.137 DEBUG    Loaded group: os_windows
   11.137 DEBUG    Adding child group os_windows to parent all
   11.137 DEBUG    Adding host USERTSTUBUNTU1_2bd7 to group os_linux
   11.137 DEBUG    Adding child group os_linux to parent all
   11.137 DEBUG    Adding host USERTSTWINDOWS1_4df5 to group os_windows
   11.137 DEBUG    Adding child group os_windows to parent all
   11.137 INFO     Loaded 3 groups, 2 hosts
   11.185 DEBUG    Inventory variables unmodified
   11.187 DEBUG    Group "os_linux" variables unmodified
   11.188 DEBUG    Group "os_windows" variables unmodified
   11.188 DEBUG    Group "skEnvironmentType_EnvNotDefined" variables unmodified
   11.195 DEBUG    Host "USERTSTUBUNTU1_2bd7" variables unmodified
   11.195 DEBUG    Host "USERTSTWINDOWS1_4df5" variables unmodified
   11.206 DEBUG    Host "USERTSTUBUNTU1_2bd7" already in group "os_linux"
   11.213 DEBUG    Host "USERTSTWINDOWS1_4df5" already in group "os_windows"
   11.221 DEBUG    Host "USERTSTUBUNTU1_2bd7" already in group "skEnvironmentType_EnvNotDefined"
   11.221 DEBUG    Host "USERTSTWINDOWS1_4df5" already in group "skEnvironmentType_EnvNotDefined"
2020-12-04 19:33:04,578 DEBUG    awx.main.models.inventory Going to update inventory computed fields, pk=9
2020-12-04 19:33:04,596 DEBUG    awx.main.models.inventory Finished updating inventory computed fields, pk=9, in 0.016 seconds
   11.245 INFO     Inventory import completed for aes_coe_azurevms_source in 9.1s
```

##### OS / ENVIRONMENT

Cent OS 7 (running ansible and awx)

##### STEPS TO REPRODUCE

1. Use the following inventory plugin to generate the inventory.
2. The  VM that is in Azure must have a tag defined named "ansible_hostname" and should not have a tag called "ansible_ignore"
2. Set the environment variables to access Azure

```yaml
plugin: azure_rm

auth_source: env
batch_fetch: yes
cert_validation_mode: ignore

hostnames:
- tags:ansible_hostname

keyed_groups:
- prefix: skEnvironmentType
  key: tags.sk_environment_type | default('EnvNotDefined')

conditional_groups:
  os_linux: "'linux' in os_profile.system"
  os_windows: "'windows' in os_profile.system"

hostvar_expressions:
  my_host_var:
  ansible_host: private_ipv4_addresses | first

exclude_host_filters:
- tags.ansible_ignore is undefined
```

##### EXPECTED RESULTS

The abve plugin will pull the correct inventory but the host in the inventory will not get the hostname we defined in "hostnames" field using jinja2.

##### ACTUAL RESULTS

The output I pasted above should suffice.

```paste below

```

**Comment by Fred-sun** [comment]
@verodhi Thank you for reporting this issue. Can you upgrade the ansible version to 2.10? Because the version before 2.9 uses the azure_rm plugin under ansible. Thank you

**Comment by Fred-sun** [comment]
@verodhi It should have been fixed by #105 and #261, can you try it? If success, I'll close it! Thank you very much!

**Comment by verodhi** [comment]
Thank you @Fred-sun
